### ISSUE TITLE
Make Users API follow proper json spec

### ISSUE THREAD

**Issue by niranjan94** [fossasia/open-event-orga-server#3682]
- attribute names must be hyphen separated. should be `is-verified` instead of `is_verified`
- ~default sort should be ascending. (`?sort=name`). Currently it seems to be descending ([spec](http://jsonapi.org/format/#fetching-sorting))~
- Hybrid properties throw error in sorting.

The rest seem to work nice. Haven't tested relationships yet since we don't have any

**Comment by shubham-padia** [comment]
@niranjan94 Can you please recheck `default sort should be ascending. (?sort=name). Currently it seems to be descending` once, seems to be ascending for me.

**Comment by niranjan94** [comment]
For `https://api.eventyay.com/v1/users?sort=firstname` this is the response I get. Which is descending right ? But it should be ascending.

```js
{
  "meta": {
    "count": 4
  },
  "data": [
    {
      "attributes": {
        "created_at": "2016-11-29T21:58:04.939475+00:00",
        "google": null,
        "instagram": null,
        "firstname": "Saptak",
        "lastname": "Sengupta",
        "twitter": "https://twitter.com/Saptak013",
        "signup_at": "2016-11-29T22:34:01.711024+00:00",
        "is_super_admin": false,
        "thumbnail": "",
        "email": "saptak013@gmail.com",
        "avatar": "https://lh4.googleusercontent.com/-qipyOJD33AM/AAAAAAAAAAI/AAAAAAAABu4/fEO8ZwD9T3M/photo.jpg",
        "contact": "",
        "is_admin": true,
        "details": "",
        "last_accessed_at": "2017-01-09T09:17:49.293955+00:00",
        "small": "",
        "facebook": null,
        "deleted_at": null,
        "is_verified": true,
        "avatar_uploaded": "https://storage.googleapis.com/eventyay.com//image/1483456983.33.jpg/TXFQd2hMNk/1483456983.33.jpg",
        "icon": ""
      },
      "type": "user",
      "id": "9",
      "links": {
        "self": "/v1/users/9"
      }
    },
    {
      "attributes": {
        "created_at": "2016-11-26T13:44:20.969190+00:00",
        "google": "",
        "instagram": "",
        "firstname": "Staging Admin",
        "lastname": "Super Administrator",
        "twitter": "",
        "signup_at": "2016-11-26T13:44:24.572425+00:00",
        "is_super_admin": true,
        "thumbnail": null,
        "email": "eventyay@fossasia.org",
        "avatar": null,
        "contact": "",
        "is_admin": true,
        "details": "",
        "last_accessed_at": "2017-05-28T16:52:15.428166+00:00",
        "small": null,
        "facebook": "",
        "deleted_at": null,
        "is_verified": true,
        "avatar_uploaded": null,
        "icon": null
      },
      "type": "user",
      "id": "1",
      "links": {
        "self": "/v1/users/1"
      }
    },
    {
      "attributes": {
        "created_at": "2017-05-21T12:41:35.031136+00:00",
        "google": null,
        "instagram": null,
        "firstname": null,
        "lastname": null,
        "twitter": null,
        "signup_at": "2017-05-21T14:15:51.997572+00:00",
        "is_super_admin": false,
        "thumbnail": null,
        "email": "jd@as.com",
        "avatar": null,
        "contact": null,
        "is_admin": false,
        "details": null,
        "last_accessed_at": null,
        "small": null,
        "facebook": null,
        "deleted_at": null,
        "is_verified": false,
        "avatar_uploaded": null,
        "icon": null
      },
      "type": "user",
      "id": "60",
      "links": {
        "self": "/v1/users/60"
      }
    },
    {
      "attributes": {
        "created_at": "2017-01-18T06:10:47.747396+00:00",
        "google": null,
        "instagram": null,
        "firstname": null,
        "lastname": null,
        "twitter": null,
        "signup_at": "2017-01-18T20:35:35.062190+00:00",
        "is_super_admin": false,
        "thumbnail": null,
        "email": "hp@fossasia.org",
        "avatar": null,
        "contact": null,
        "is_admin": true,
        "details": null,
        "last_accessed_at": "2017-01-20T16:20:37.971012+00:00",
        "small": null,
        "facebook": null,
        "deleted_at": null,
        "is_verified": true,
        "avatar_uploaded": null,
        "icon": null
      },
      "type": "user",
      "id": "54",
      "links": {
        "self": "/v1/users/54"
      }
    }
  ],
  "jsonapi": {
    "version": "1.0"
  },
  "links": {
    "self": "/v1/users?sort=firstname"
  }
}
```

**Comment by niranjan94** [comment]
Okay. Ignore my comment. It is in ascending. Me blind 😋

**Comment by shubham-padia** [comment]
There's also another bug, Hybrid properties throw error in sorting, looking into it.

**Comment by shubham-padia** [comment]
`attribute names must be hyphen separated. should be is-verified instead of is_verified`, how much of a blocker is this ? `flask-rest-jsonapi` doesn't play well with hyphens.

**Comment by niranjan94** [comment]
Hmm... Okay. This can be compensated for in client side. But just
curious... What issue does the lib have with hyphen separated attribute
names?

**Comment by shubham-padia** [comment]
@niranjan94 As variables can't have hyphens, marshmallow provides http://marshmallow.readthedocs.io/en/latest/api_reference.html#module-marshmallow.fields, load_from and dump_to, but weirdly flask-rest-jsonapi doesn't respect them at some places. If I give a query with `?sort=created-at` it interprets the sorting param as createdat and throws an error.

**Comment by niranjan94** [comment]
Hmm... Okay... Weird. Cool. This shouldn't be a big blocker. Can be
corrected in the client side pretty easily. Will confirm though.

**Comment by shubham-padia** [comment]
Closed via #3683
